### PR TITLE
Fix Playwright setup on Gitpod

### LINKED ISSUE #2257: Unable to connect to DB in gitpod - tests failing
### Description

When running tests in gitpod in the browser IDE, tests fail due to no DB connection.  Suspect a configuration issue.

**Steps to recreate:**
1. create a branch
2. run project in gitpod
3. in terminal, run command: npm test

See error in SS below

db_connect_error

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dan-mba**
## Fixes Issue

Closes #2257

## Changes proposed

Install playwright dependencies on init.
Set Playwright to not run parallel tests on Gitpod 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2261"></a>

### CODE REVIEW

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

File: .gitpod.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ---
 # List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
 tasks:
-  - name: Install npm depdencies
-    init: npm i
+  - name: Install npm dependencies
+    init: |
+      npm ci
+      npx playwright install --with-deps chromium
```
Comment (eddiejaoude) [inline .gitpod.yml:7]: It didn't work for me, I still needed to run this command manually to make the test pass 🤔 

I wonder why this command doesn't run when building the project on GitHub

**Comment by Cahllagerfeld** [comment]
The scripts need to be executed in a custom-dockerfile. 
The init task only persists the data inside the `workspace` directory.
Depending if the tests should be able to run headed, the base image of the dockerfile needs to be `gitpod/workspace-full-vnc`

**Comment by dan-mba** [comment]
> The scripts need to be executed in a custom-dockerfile. The init task only persists the data inside the `workspace` directory. Depending if the tests should be able to run headed, the base image of the dockerfile needs to be `gitpod/workspace-full-vnc`

Would the playwright install be the only command needed in the dockerfile or would the `npm ci` be needed in there too?
Does it make sense to support headed tests or Gitpod, or would it make more sense to to have people run that elsewhere if needed?

**Comment by eddiejaoude** [comment]
How do we do it for the `finder` project? Could we do it similar 
https://github.com/EddieHubCommunity/good-first-issue-finder/blob/main/.gitpod.yml

**Comment by dan-mba** [comment]
I think I have it working now.

I was able to install the OS level browser dependencies in the Dockerfile.
But I had to install Chromium in the gitpod.yml command section in order to make it work consistently.

File: .gitpod.yml [hunk 2]
```diff
@@ -1,9 +1,11 @@
 ---
 # List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
 tasks:
-  - name: Install npm depdencies
-    init: npm i
+  - name: Install npm dependencies
+    init: |
+      npm ci
     command: |
+      npx playwright install chromium
```
Comment (Cahllagerfeld) [inline .gitpod.yml:8]: When running this command in the dockerfile, we shouldn't need it here

**Comment by eddiejaoude** [comment]
Thank you both for continuing to work through this 💪

File: .gitpod.yml (hunk 2 above)
Comment (Cahllagerfeld) [reply .gitpod.yml:8]: Oh I see 👀 
An assumption from my side - I didn't test it. Could we potentially run `npx playwright install --with-deps chromium` in the dockerfile and it takes care of all?

Comment (dan-mba) [reply .gitpod.yml]: I found a way to install chromium in `node_modules` via the [Playwright docs](https://playwright.dev/docs/browsers#hermetic-install).

I updated the `.gitpod.yml` to do this in the init section.

**Comment by eddiejaoude** [comment]
It works 🎉 

```bash
gitpod /workspace/LinkFree (gitpod) $ npm run test

> linkfree@1.9.0 test
> npx playwright test


Running 19 tests using 1 worker
[WebServer] (node:1331) [DEP0128] DeprecationWarning: Invalid 'main' field in '/workspace/LinkFree/node_modules/react-icons/package.json' of 'lib'. Please either fix that or report it to the module author
(Use `node --trace-deprecation ...` to show where the warning was created)
[WebServer] Warning: data for page "/search" (path "/search?username=_test-profile-does-not-exist") is 492 kB which exceeds the threshold of 128 kB, this amount of data can reduce performance.
See more info here: https://nextjs.org/docs/messages/large-page-data
[WebServer] Warning: data for page "/search" is 492 kB which exceeds the threshold of 128 kB, this amount of data can reduce performance.
See more info here: https://nextjs.org/docs/messages/large-page-data
[19/19] [chromium] › search.spec.js:44:6 › find the profile after providing concise name

  10 skipped
  9 passed (43s)

To open last HTML report run:

  npx playwright show-report
```

**Review by eddiejaoude — APPROVED** [review]
wow great work Dan 🔥 

Thanks @Cahllagerfeld for supporting with this 👍

**Review by Panquesito7 — APPROVED** [review]
Looks great! Thanks. 🚀

**Pull request merged**